### PR TITLE
[Java] Archive read/write counters.

### PR DESCRIPTION
**Pull request by vyazelenko**
This PR adds three counters for recorder and replayer:
* total number of bytes written/read
* total time for write/read in ns
* max write/read time in ns for a signle block of data

Sample output from:
```
 40:        6,400,000,000 - archive-recorder total write bytes
 41:        1,588,068,973 - archive-recorder total write time in ns
 42:              600,904 - archive-recorder max write time in ns
 43:       19,200,000,000 - archive-replayer total read bytes
 44:        2,240,055,098 - archive-replayer total read time in ns
 45:            2,480,968 - archive-replayer max read time in ns
```

### CODE REVIEW

**Comment by tmontgomery** [comment]
Looks pretty good to me.

**Review by tmontgomery — APPROVED** [review]

File: aeron-archive/src/main/java/io/aeron/archive/ArchiveConductor.java [hunk 1]
```diff
@@ -2111,4 +2122,64 @@ private void deleteSegments(
...
+            totalWriteTimeCounter = context.totalWriteTimeCounter();
+            maxWriteTimeCounter = context.maxWriteTimeCounter();
+        }
+
+        public int doWork()
+        {
+            final int workCount = super.doWork();
+            if (workCount > 0)
+            {
+                totalWriteBytesCounter.set(totalWriteBytes.get());
```
Comment (mjpt777) [inline aeron-archive/src/main/java/io/aeron/archive/ArchiveConductor.java:2148]: Should this be setOrdered as a release is only needed?

File: aeron-client/src/main/java/io/aeron/AeronCounters.java [hunk 2]
```diff
@@ -160,6 +160,42 @@
...
+    public static final int ARCHIVE_RECORDER_TOTAL_WRITE_BYTES_TYPE_ID = 106;
+
+    /**
+     * The type id of the {@link Counter} used for keeping track of the total time the recorder spent writing data to
+     * the storage.
+     */
+    public static final int ARCHIVE_RECORDER_TOTAL_WRITE_TIME_TYPE_ID = 107;
+
+    /**
+     * The type id of the {@link Counter} used for keeping track of the max it took replayer to read a block from the
```
Comment (mjpt777) [inline aeron-client/src/main/java/io/aeron/AeronCounters.java:182]: I think a "time" is missing after max.

**Comment by mjpt777** [comment]
LGTM with some minor tweaks.

File: aeron-archive/src/main/java/io/aeron/archive/ArchiveConductor.java (hunk 1 above)
Comment (vyazelenko) [reply aeron-archive/src/main/java/io/aeron/archive/ArchiveConductor.java:2148]: Done.

File: aeron-client/src/main/java/io/aeron/AeronCounters.java (hunk 2 above)
Comment (vyazelenko) [reply aeron-client/src/main/java/io/aeron/AeronCounters.java]: Fixed.

**Pull request merged**